Library/Homebrew/completions.rb

```diff
@@ -296,7 +296,15 @@ def generate_fish_subcommand_completion(command)
           next unless FISH_NAMED_ARGS_COMPLETION_FUNCTION_MAPPING.key? type
 
           named_arg_function = FISH_NAMED_ARGS_COMPLETION_FUNCTION_MAPPING[type]
-          named_args << "__fish_brew_complete_arg '#{command}' -a '(#{named_arg_function})'"
+          named_arg_prefix = "complete -f -c brew -n '__fish_brew_command #{command}; and not __fish_seen_argument"
+
+          named_args << if type == :formula && named_args_types.include?(:cask)
```

Comment:
In addition to the `:formula` and `:cask` types, there are also `:installed_{formula,cask}` and `:outdated_{formula,cask}` which should get the same logic here. This will make the completions work for commands like `uninstall` and `upgrade`, too.

Reply:
Makes sense! I will add this later.